qiskit_experiments/library/characterization/analysis/ramsey_xy_analysis.py

```diff
@@ -422,6 +429,69 @@ def _default_options(cls):
 
         return options
 
+    def _format_data(
+        self,
+        curve_data: curve.ScatterTable,
+    ) -> curve.ScatterTable:
+        columns = list(curve_data.columns)
+
+        formatted_data = super()._format_data(curve_data)
+        raw_ramsey = formatted_data.filter(like="formatted", axis="index")
+
+        # Convert to phase format with unwrapper
+        phase_data = np.empty((0, len(columns)))
+        y_mean = raw_ramsey.yval.mean()
+        grouped = raw_ramsey.groupby(["direction", "series"])
+
+        for m_id, direction in enumerate(("pos", "neg")):
+            x_quadrature = grouped.get_group((direction, "X"))
+            y_quadrature = grouped.get_group((direction, "Y"))
+            if not np.array_equal(x_quadrature.xval, y_quadrature.xval):
+                raise ValueError(
+                    "Amplitude values of X and Y quadrature are different. "
+                    "Same values must be used."
+                )
+            x_uarray = unp.uarray(x_quadrature.yval, x_quadrature.yerr)
+            y_uarray = unp.uarray(y_quadrature.yval, y_quadrature.yerr)
+
+            amplitudes = x_quadrature.xval.to_numpy()
+
+            # pylint: disable=no-member
+            phase = unp.arctan2(y_uarray - y_mean, x_uarray - y_mean)
+            phase_n = unp.nominal_values(phase)
+            phase_s = unp.std_devs(phase)
+
+            # Unwrap phase
+            # We assume a smooth slope and correct 2pi phase jump to minimize the change of the slope.
+            if amplitudes[0] < 0:
+                # Flip array order because this is array is negative increments.
+                phase_n = phase_n[::-1]
```

Comment:
Yes, it's bidirectional but accumulates the y value from zero index. On the negative side, zero index value corresponds to the maximum absolute shift, and starting from there will cause nonzero phase at x=0 on negative side, i.e. discontinuity. This offset cannot be simply subtracted because this consists of the offset from unwrapping and some physical offset from frequency mis-calibration.

Reply:
Ah, okay. I think an alternative would be:

```python
            unwrapped_phase = np.unwrap(phase_n)
            if amplitudes[0] < 0:
                # Preserve phase value closest to 0 amplitude
                unwrapped_phase = unwrapped_phase + (phase_n[-1] - unwrapped_phase[-1])
```

I don't know that that is any clearer though. I think we could just make the current comment clearer.

Reply:
This looks cleaner. Done in [cd72c27](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1243/commits/cd72c278fe3e4f5e4400e53297ae9b3f998d9cd2)